osxkeychain.go

```diff
@@ -137,37 +138,48 @@ func authenticationTypeToGo(authtype C.CFTypeRef) AuthenticationType {
 
 // Adds an Internet password to the user's default keychain.
 func AddInternetPassword(pass *InternetPassword) error {
+	// TODO: Check fields for UTF-8 encoding and size fitting in
+	// 32 bits.
+
+	serverName := C.CString(pass.ServerName)
+	defer C.free(unsafe.Pointer(serverName))
+
+	// TODO: Make optional.
```

Comment:
Did what you suggested. I tried to find out what SecurityDomain actually is, but had no luck. I don't think there's any harm in assuming the empty string means NULL.